### ISSUE TITLE
Q: How can I improve some of the UI text?

### ISSUE THREAD

**Issue by davelab6** [fontforge/designwithfontforge.com#20]
Question: How can I improve some of the UI text?

Answer:

In September 2013 Ben Crowder said on the users mailing list, 

> In preferences dialog, just putting a space in between each word instead of camel case would be great. "Undo Redo Limit To Save" instead of "UndoRedoLimitToSave", for example.

You don't need to be a C programmer to make such a change, just know web development and a little UNIX :)

First, make a fork of FontForge (https://github.com/fontforge/fontforge) in the usual Github way, and download your fork.

Then search the codebase for the text you want to fix:

``` sh
$ grep -r UndoRedoLimitToSave *
fontforge/prefs.c:extern int UndoRedoLimitToSave;  /* sfd.c */
fontforge/prefs.c:    { N_("UndoRedoLimitToSave"), pr_int, &UndoRedoLimitToSave, NULL, NULL, '\0', NULL, 0, N_( "The number of undo and redo operations which will be saved in sfd files.\nIf you set this to 0 undo/redo information is not saved to sfd files.\nIf set to -1 then all available undo/redo information is saved without limit.") },
fontforge/sfd.c:int UndoRedoLimitToSave = 0;
fontforge/sfd.c:        if( saveUndoes && UndoRedoLimitToSave > 0) {
fontforge/sfd.c:                for( limit = UndoRedoLimitToSave;
fontforge/sfd.c:                limit = UndoRedoLimitToSave;
fontforge/sfd.c:                for( limit = UndoRedoLimitToSave;
fontforge/sfd.c:    if( !UndoRedoLimitToSave )
po/ja.po:msgid "UndoRedoLimitToSave"
po/pl.po:msgid "UndoRedoLimitToSave"
po/ru.po:msgid "UndoRedoLimitToSave"
po/uk.po:msgid "UndoRedoLimitToSave"
$
```

From this we can see that the prefs.c file has a part that is a translatable 'msgid' - N_("UndoRedoLimitToSave") - and there are translation source files as *.po with the msgid.
1. Open prefs.c and change N_("UndoRedoLimitToSave") to N_("Undo Redo Limit ToSave")
2. Open the po files and change msgid "UndoRedoLimitToSave" to msgid "Undo Redo Limit To Save"
3. Save these files, and test them. The easiest way to do this on Mac OS X is to commit the changes and push them to your repo, then edit the Homebrew formula :) 

``` sh
$ brew edit fontforge
```

Then replace what is currently line 8, 

``` ruby
  head 'https://github.com/fontforge/fontforge.git'
```

with your github url, save the file, and

``` sh
brew uninstall fontforge; 
brew install fontforge --HEAD --with-x
```
1. When you are happy with your changes, make a pull request in the normal Github way.

**Comment by bencrowder** [comment]
Question: are there compilation instructions for OS X? Right now I use HomeBrew to compile the HEAD, but I'm not sure how to point HomeBrew to compile my fork instead. (I tried compiling it outside of HomeBrew but I'm getting environment-related errors, and I'd like to compile using HomeBrew's settings so I know my changes actually work with the normal setup.)

**Comment by davelab6** [comment]
The compilation instructions are the Homebrew formula :) 

``` sh
$ brew edit fontforge
```

Then replace what is currently line 8, 

``` ruby
  head 'https://github.com/fontforge/fontforge.git'
```

with your github url, save the file, and `brew uninstall fontforge; brew install fontforge --HEAD --with-x`

I update the issue text to reflect this important testing step :D

**Comment by bencrowder** [comment]
Awesome, thanks.

**Comment by vernnobile** [comment]
any idea how to force homebrew to use nano instead of vim to edit ?? :)
i effing hate vim.

**Comment by davelab6** [comment]
@vernnobile https://www.google.com/search?q=unix+shell+editor+variable

**Comment by vernnobile** [comment]
ahh thanks :) Homebrew uses the system default.
To change OSX default from vim to nano, i ran;
EDITOR=nano; export EDITOR

ps

brew install fontforge --HEAD --with-x

fails for me on 10.8.4 :(

iMac:~ vern$ brew install -v fontforge --HEAD --with-x
==> Cloning https://github.com/fontforge/fontforge.git
git status -s
Updating /Library/Caches/Homebrew/fontforge--git
git config remote.origin.url https://github.com/fontforge/fontforge.git
git config remote.origin.fetch +refs/heads/master:refs/remotes/origin/master
git fetch origin
git reset --hard origin/HEAD
HEAD is now at 39e86e0 Update prefs to ensure Snap to Int is on by default
git checkout-index -a -f --prefix=/private/tmp/fontforge-xbB0/
==> ./configure --prefix=/usr/local/Cellar/fontforge/HEAD --enable-double --without-freetype-bytecode --with-python-headers=/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 --with-python-lib=-lpython2.7 --enable-pyextension --without-cairo --without-pango
./configure --prefix=/usr/local/Cellar/fontforge/HEAD --enable-double --without-freetype-bytecode --with-python-headers=/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 --with-python-lib=-lpython2.7 --enable-pyextension --without-cairo --without-pango
Error: Failure while executing: ./configure --prefix=/usr/local/Cellar/fontforge/HEAD --enable-double --without-freetype-bytecode --with-python-headers=/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 --with-python-lib=-lpython2.7 --enable-pyextension --without-cairo --without-pango

**Comment by davelab6** [comment]
brew install fontforge --HEAD --with-x --debug

and try to rerun the configure manually from the shell

**Comment by khaledhosny** [comment]
There is a (technical) reason those names has no spaces (written in a comment somewhere in the code, I do not remember), so you need to fix that first, not just blindly insert spaces.

**Comment by bencrowder** [comment]
Yep, I'm aware of that. (The comment -- by George -- says that spaces should be inserted and explains how to do it, which is the way Dave recommends above.)
